crates/cli/cli/src/commands/query/connection/end.rs

```diff
@@ -49,43 +46,23 @@ pub struct QueryConnectionEnd {
 impl CommandRunner<CosmosBuilder> for QueryConnectionEnd {
     async fn run(&self, builder: &CosmosBuilder) -> Result<Output> {
         let chain = builder.build_chain(&self.chain_id).await?;
-        let chain_id = self.chain_id.clone();
-        let connection_id = self.connection_id.clone();
-        let height = self.height;
-
-        let connection_end = chain
-            .with_blocking_chain_handle(move |chain_handle| {
-                let query_height = if let Some(height) = height {
-                    let specified_height = Height::new(chain_handle.id().version(), height)
-                    .map_err(|e| RelayerError::generic(eyre!("Failed to create Height with revision number `{}` and revision height `{height}`. Error: {e}", chain_handle.id().version())))?;
+        let height = self.height.map(|h| Height::new(chain.chain_id().version(), h).map_err(|e| RelayerError::generic(eyre!("Failed to create Height with revision number `{}` and revision height `{h}`. Error: {e}", chain.chain_id().version()))).unwrap());
```

Comment:
There's something weird going on with the formatting here

Reply:
Yes I've had odd behaviour with `cargo fmt` and the CI doesn't seem to complain either. I don't know what's going on

Reply:
As last resort you can mark the block with `#[rustfmt::skip]` and format it manually.

Reply:
There is a `.map_err()` followed by an `.unwrap()`, which doesn't really make sense. And if you want to do `map_err()`, perhaps it is better to do it right after `Height::new()`.

Reply:
> As last resort you can mark the block with #[rustfmt::skip] and format it manually.

The lines inside the `.map` don't seem to be impacted by `cargo fmt`, I can format it manually without disabling rustfmt

> There is a .map_err() followed by an .unwrap(), which doesn't really make sense. And if you want to do map_err(), perhaps it is better to do it right after Height::new().

Oh good catch. I will update this using `transpose()` and `?` instead of the `unwrap`. But